src/test/java/org/broadinstitute/hellbender/tools/walkers/coverage/DepthOfCoverageIntegrationTest.java

```diff
@@ -0,0 +1,259 @@
+        final File output = IOUtils.createTempFileInDirectory( "testNoCoverageDueToFiltering", ".csv", baseOutputFile);
+
+        String cmd = "-R "+hg38Reference+" " +
+        "-I "+largeFileTestDir+"multiSampleSubsetted.bam"+
+        " -O "+output.getAbsolutePath() + " -L chr1:1570000-1571000 --omit-interval-statistics --omit-locus-table --omit-per-sample-statistics";
+        runCommandLine(cmd.split(" "));
+        File[] actualFiles = baseOutputFile.listFiles();
+
+        compareOutputDirectories(expectedBaseName, output.getName(), actualFiles);
+    }
+
+    @Test
+    // Coverage drops to nothing because '--max-base-quality 4' is set, resuling in very poor output coverage
+    public void testNoCoverageDueToFiltering() throws IOException {
+        final String expectedBaseName = "depthofcoveragenwithiltering";
+        final File baseOutputFile = createTempDir("testNoCoverageDueToFiltering");
+        final File output = IOUtils.createTempFileInDirectory( "testNoCoverageDueToFiltering", ".csv", baseOutputFile);
+
+        String cmd = "-R "+hg38Reference+" " +
+                "-I "+largeFileTestDir + "multiSampleSubsetted.bam " +
+                "-L "+ getTestFile("artificial.target_region.interval_list ")+
+                "--min-base-quality 5 --max-base-quality 4 -dels --print-base-counts -pt readgroup -pt sample -pt library -pt platform --output-format CSV";
+        cmd += " -O "+output.getAbsolutePath();
+        runCommandLine(cmd.split(" "));
+
+        File[] actualFiles = baseOutputFile.listFiles();
+
+        compareOutputDirectories(expectedBaseName, output.getName(), actualFiles);
+    }
+
+    @Test
+    // NOTE, the gene list file was not generated with GATK3 due to different gene list merging behavior
+    public void testGeneListDataAllCovered() throws IOException {
+        final String expectedBaseName = "testGeneListDataAllCovered";
+        final File baseOutputFile = createTempDir("testNoCoverageDueToFiltering");
+        final File output = IOUtils.createTempFileInDirectory( "testNoCoverageDueToFiltering", ".csv", baseOutputFile);
+        output.delete();
+
+        String[] cmd = new String[]{ "-R",hg38Reference,
+                "-I",largeFileTestDir+"multiSampleSubsetted.bam",
+                "-L",getTestFile("artificial.gene_target.interval_list").getAbsolutePath(),
+                "--calculate-coverage-over-genes",getTestFile("refGene_CDK11B.refseq").getAbsolutePath(),
+                "--min-base-quality","0","-dels","-pt","sample","--output-format","CSV","--omit-depth-output-at-each-base","--omit-per-sample-statistics",
+                "-O",output.getAbsolutePath()};
+        runCommandLine(cmd);
+
+        File[] actualFiles = baseOutputFile.listFiles();
+
+        compareOutputDirectories(expectedBaseName, output.getName(), actualFiles);
+    }
+
+    @Test
+    public void testIntervalListOverlappingUniqueBehavior() throws IOException {
```

Comment:
What does "unique behavior" mean? Add a comment explaining what this test case is testing.